boards/BMS/Src/App/states/App_InitState.c

```diff
@@ -30,24 +32,32 @@ static void InitStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
     if (App_AllStatesRunOnTick100Hz(state_machine))
     {
-        struct BmsWorld *world = App_SharedStateMachine_GetWorld(state_machine);
+        struct BmsWorld *      world   = App_SharedStateMachine_GetWorld(state_machine);
+        struct TractiveSystem *ts      = App_BmsWorld_GetTractiveSystem(world);
+        struct Airs *          airs    = App_BmsWorld_GetAirs(world);
+        struct Charger *       charger = App_BmsWorld_GetCharger(world);
+
+        bool is_charger_connected = App_Charger_IsConnected(charger);
+
 #ifndef BSPD_DEMO_MODE
         struct TractiveSystem *ts   = App_BmsWorld_GetTractiveSystem(world);
         struct Airs *          airs = App_BmsWorld_GetAirs(world);
         // don't allow pre_charge if in BSPD_DEMO_MODE
         if (App_Airs_IsAirNegativeClosed(airs) && (App_TractiveSystem_GetVoltage(ts) < TS_DISCHARGED_THRESHOLD_V))
         {
-            App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
+            // If the charger is connected, the CAN message must also be set
+            // to continue into Pre-Charge State, then Charge State
+            if (is_charger_connected && App_CanRx_DEBUG_ChargingSwitch_StartCharging_Get())
+            {
+                App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
+            }
+            // If charger isn't connected, go into Pre-Charge State, then Drive State
+            else if (!is_charger_connected)
+            {
+                App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
+            }
```

Comment:
I feel like the use of an if/else here is confusing since both conditions lead to precharge state. 
Consider changing to a single if statement, something like:
```c
bool precharge_for_charging = is_charger_connected && App_CanRx_DEBUG_ChargingSwitch_StartCharging_Get();
if(precharge_for_charging || !is_charger_connected)
{
    App_SharedStateMachine_SetNextState(state_machine, App_GetPreChargeState());
}
```
```